MatchGame/Deck.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MatchGame
+{
+    public class Deck : IDeck
+    {
+        private List<Card> _cards;
```

Comment:
Why list? Yes we care about order/postioning, but remember, depend on abstractions, not implementations.